vignettes/articles/identifying-false-positives.Rmd

```diff
@@ -0,0 +1,409 @@
+
+There are, of course, other factors that can help to validate the detection. We will examine two that we expect to be explanatory, noisiness and continuity, and create additional variables to describe these factors. We consider times when there are many recordings at a given receiver to be noisy. We can use the `activity` table to identify the number of runs and their lengths for each antenna of each receiver recorded per hour. We can get a more complete picture of noisiness using the `activity` table because it includes numbers of runs across all projects, not just our own. We can also calculate ‘proportion of short run lengths in runs per receiver hour’ and link the data from the `activity` table back to the runs in the `alltags` table via the `runs` table by calculating the `hourBin` for each `runID`. This allows us to connect the information on noise in the `activity` table back to our detection data. In addition, there might be a propensity for some tags to be recorded with short run lengths, i.e. detections resulting from noise. We can therefore calculate for each tag a variable for the overall ‘proportion of short run lengths’.
+
+```{r, warning = FALSE}
+
+# get activity table
+a <- tbl(db, "activity")
+
+adf <- a %>% 
+  collect %>% as.data.frame                                               # convert into flat data frame
+  
+rm(a)
+
+# get runs table
+r <- tbl(db, "runs")
+
+rdf <- r %>% 
+  collect %>% as.data.frame %>% 
+  mutate(hourBin = floor(tsBegin/3600))                                # calculate hourBin for each run
+
+rm(r)
+
+# join runs and activity tables to get hourly values on runs/recv for each runID
+radf <- left_join(rdf, adf, by = c("batchIDbegin" = "batchID", "ant", "hourBin"))
+
+# join with detection data
+df <- left_join(df, radf %>% select(runID, numRuns, numHits, run2, run3), by = "runID")
```

Comment:
No need to pipe and when inside another function a bit easier to read if we avoid it
```suggestion
df <- left_join(df, select(radf, runID, numRuns, numHits, run2, run3), by = "runID")
```